pkg/controller/util/cluster_util.go

```diff
@@ -101,3 +115,119 @@ func IsPrimaryCluster(obj, clusterObj pkgruntime.Object) bool {
 	clusterMeta := MetaAccessor(clusterObj)
 	return meta.GetUID() == clusterMeta.GetUID()
 }
+
+// CustomizeTLSTransport replaces the restclient.Config.Transport with one that
+// implements the desired TLS certificate validations
+func CustomizeTLSTransport(fedCluster *fedv1b1.KubeFedCluster, clientConfig *restclient.Config) error {
+	clientTransportConfig, err := clientConfig.TransportConfig()
+	if err != nil {
+		return errors.Errorf("Cluster %s client transport config error: %s", fedCluster.Name, err)
+	}
+	transportConfig, err := transport.TLSConfigFor(clientTransportConfig)
+	if err != nil {
+		return errors.Errorf("Cluster %s transport error: %s", fedCluster.Name, err)
+	}
+
+	err = CustomizeCertificateValidation(fedCluster, transportConfig)
+	if err != nil {
+		return errors.Errorf("Cluster %s custom certificate validation error: %s", fedCluster.Name, err)
+	}
+
+	// using the same defaults as http.DefaultTransport
```

Comment:
Why not just copy `http.DefaultTransport` and then set the specific values you're interested in?

Reply:
Kubernetes go-client creates a new Transport using the same values as DefaultTransport to avoid conflicts between different clients.  I am employing the same pattern since we have to create the transport in case of custom TLS options.

Some backing documentation:
https://github.com/hashicorp/go-cleanhttp

> The Go standard library contains a default http.Client called http.DefaultClient. It is a common idiom in Go code to start with http.DefaultClient and tweak it as necessary, and in fact, this is encouraged; from the http package documentation:
> 
> > The Client's Transport typically has internal state (cached TCP connections), so Clients should be reused instead of created as needed. Clients are safe for concurrent use by multiple goroutines.
> 
> 
> Unfortunately, this is a shared value, and it is not uncommon for libraries to assume that they are free to modify it at will. With enough dependencies, it can be very easy to encounter strange problems and race conditions due to manipulation of this shared value across libraries and goroutines (clients are safe for concurrent use, but writing values to the client struct itself is not protected).
> 
> Making things worse is the fact that a bare http.Client will use a default http.Transport called http.DefaultTransport, which is another global value that behaves the same way. So it is not simply enough to replace http.DefaultClient with &http.Client{}.